examples/resnet/resnet_main.py

```diff
@@ -132,36 +139,50 @@ def accuracy(self, weights):
 
     precision = 1.0 * correct_prediction / total_prediction
     self.best_precision = max(precision, self.best_precision)
+    precision_summ = tf.Summary()
+    precision_summ.value.add(
+        tag='Precision', simple_value=precision)
+    self.summary_writer.add_summary(precision_summ, train_step)
+    best_precision_summ = tf.Summary()
+    best_precision_summ.value.add(
+        tag='Best Precision', simple_value=self.best_precision)
+    self.summary_writer.add_summary(best_precision_summ, train_step)
+    self.summary_writer.add_summary(summaries, train_step)
+    tf.logging.info('loss: %.3f, precision: %.3f, best precision: %.3f' %
+                    (loss, precision, self.best_precision))
+    self.summary_writer.flush()
     return precision
 
+  def get_ip_addr(self):
+    return self.ip_addr
+
 def train():
   """Training loop."""
   num_gpus = int(FLAGS.num_gpus)
-  ray.init(num_gpus=num_gpus)
-  train_data = get_data.remote(FLAGS.train_data_path, 50000)
-  test_data = get_data.remote(FLAGS.eval_data_path, 10000)
+  ray.init(num_gpus=num_gpus, redirect_output=True)
+  train_data = get_data.remote(FLAGS.train_data_path, 50000, FLAGS.dataset)
+  test_data = get_data.remote(FLAGS.eval_data_path, 10000, FLAGS.dataset)
+  IPython.embed()
   if num_gpus > 0:
-    train_actors = [ResNetTrainActor(train_data, num_gpus) for _ in range(num_gpus)]
+    train_actors = [ResNetTrainActor(train_data, FLAGS.dataset, num_gpus) for _ in range(num_gpus)]
   else:
     train_actors = [ResNetTrainActor(train_data, num_gpus)]
-  test_actor = ResNetTestActor(test_data, 50)
+  test_actor = ResNetTestActor(test_data, FLAGS.dataset, FLAGS.eval_batch_count, FLAGS.eval_dir)
+  print('The log files for tensorboard are stored at ip {}.'.format(ray.get(test_actor.get_ip_addr())))
   step = 0
   weight_id = train_actors[0].get_weights()
-  acc_id = test_actor.accuracy(weight_id)
+  acc_id = test_actor.accuracy(weight_id, step)
   if num_gpus == 0:
     num_gpus = 1
+  print("Starting computation.")
   while True:
-    with open('results.txt', 'a') as results:
-      print('Computing steps')
-      all_weights = ray.get([actor.compute_steps(weight_id) for actor in train_actors])
-      mean_weights = {k: sum([weights[k] for weights in all_weights]) / num_gpus for k in all_weights[0]}
-      weight_id = ray.put(mean_weights)
-      step += 10
-      if step % 200 == 0:
-        acc = ray.get(acc_id)
-        acc_id = test_actor.accuracy(weight_id)
-        print('Step {0}: {1:.6f}'.format(step - 200, acc))
```

Comment:
we should still print something like this so the user knows that the computation is progressing